datasource/src/main/scala/quasar/physical/mongo/Mongo.scala

```diff
@@ -257,17 +226,6 @@ object Mongo {
     }
   }
 
-  def singleObservableAsF[F[_]: Async: ContextShift, A](obs: SingleObservable[A]): F[A] =
-    Async[F].async { cb: (Either[Throwable, A] => Unit) =>
-      obs.subscribe(new Observer[A] {
-        override def onNext(r: A): Unit = cb(Right(r))
-
-        override def onError(e: Throwable): Unit = cb(Left(e))
-
-        override def onComplete() = ()
-      })
-    } guarantee ContextShift[F].shift
```

Comment:
Do we need to shift in this impl?

Reply:
I think we're ok since `StreamSubscriber` uses `unsafeRunAsync` to avoid blocking the producer.